app/controllers/textile_controller.rb

```diff
@@ -0,0 +1,55 @@
+class TextileController < AuthenticatedController
+  # Returns a single field for the form view
+  def field
+    @index = params[:index].to_i
+  end
+
+  # Returns the form view given a source text
+  def form
+    @form_data = Note.parse_fields(params[:source]) if params[:source]
+    @allow_dropdown = JSON.parse(params[:allow_dropdown] || 'false')
+  end
+
+  # Returns the markup cheatsheet that is used by the jQuery.Textile plugin Help
+  # button.
+  def markup_help
+    render layout: false
+  end
+
+  # Returns the source text given a form data
+  def source
+    render plain: convert_to_source(form_params)
+  end
+
+  # Returns the Textile version of a text passed as parameter
+  def textilize
+    @text =
+      if params[:form]
+        convert_to_source(form_params)
+      else
+        params[:text]
```

Comment:
When is this path taken?

Reply:
This is triggered when the Source view is being used,